waspc/src/Wasp/Project/Analyze.hs

```diff
@@ -69,8 +82,109 @@ analyzeWaspProject waspDir options = do
   where
     fileNotFoundMessage = "Couldn't find the *.wasp file in the " ++ toFilePath waspDir ++ " directory"
 
-analyzeWaspFile :: Psl.Schema.Schema -> Path' Abs File' -> IO (Either [CompileError] [AS.Decl])
-analyzeWaspFile prismaSchemaAst waspFilePath = do
+data WaspFile
+  = WaspLang !(Path' Abs (File WaspLangFile))
+  | WaspTs !(Path' Abs (File WaspTsFile))
+
+data WaspLangFile
+
+data WaspTsFile
+
+data CompiledWaspJsFile
+
+data SpecJsonFile
+
+analyzeWaspFile :: Path' Abs (Dir WaspProjectDir) -> Psl.Schema.Schema -> WaspFile -> IO (Either [CompileError] [AS.Decl])
+analyzeWaspFile waspDir prismaSchemaAst = \case
+  WaspLang waspFilePath -> analyzeWaspLangFile prismaSchemaAst waspFilePath
+  WaspTs waspFilePath -> analyzeWaspTsFile waspDir prismaSchemaAst waspFilePath
+
+analyzeWaspTsFile :: Path' Abs (Dir WaspProjectDir) -> Psl.Schema.Schema -> Path' Abs (File WaspTsFile) -> IO (Either [CompileError] [AS.Decl])
+analyzeWaspTsFile waspProjectDir _prismaSchemaAst _waspFilePath = runExceptT $ do
+  -- TODO: I'm not yet sure where tsconfig.node.json location will come from
+  -- because we also need that knowledge to generate a TS SDK project.
+  compiledWaspJsFile <- ExceptT $ compileWaspTsFile waspProjectDir [relfile|tsconfig.node.json|]
+  specJsonFile <- ExceptT $ executeMainWaspJsFile waspProjectDir compiledWaspJsFile
+  contents <- ExceptT $ readDeclsJsonFile specJsonFile
+  liftIO $ putStrLn "Here are the contents of the spec file:"
+  liftIO $ print contents
+  return []
+
+-- TODO: Reconsider the return value. Can I write the function in such a way
+-- that it's impossible to get the absolute path to the compiled file without
+-- calling the function that compiles it?
+-- To do that, I'd have to craete a private module that knows where the file is
+-- and not expose the constant for creating the absoltue path (like I did with config/spec.json).
+-- Normally, I could just put the constant in the where clause like I did there, but I'm hesitant
+-- to do that since the path comes from the tsconfig.
+-- That is what I did currently, but I'll have to figure out the long-term solution.
+-- The ideal solution is reading the TS file, and passing its config to tsc
+-- manually (and getting the output file path in the process).
+compileWaspTsFile ::
+  Path' Abs (Dir WaspProjectDir) ->
+  Path' (Rel WaspProjectDir) File' ->
+  IO (Either [CompileError] (Path' Abs (File CompiledWaspJsFile)))
+compileWaspTsFile waspProjectDir tsconfigNodeFileInWaspProjectDir = do
+  chan <- newChan
+  (_, tscExitCode) <-
+    concurrently
+      (readJobMessagesAndPrintThemPrefixed chan)
+      ( runNodeCommandAsJob
```

Comment:
I'd create an issue 👍 

Different requirements for different use cases sound good to me. `RequireNode` and `RequireNpm`.

Reply:
I agree to handle it separately.
Yeah I don't think I was thinking that much about it when creating this function, the point was just, if we need to run something that relies on node and its stuff being there, we use this command. Maybe that is good enough though?

And one question: Isn't npm installed together with node? And npx with all of them together? So everything comes bundled with node? I think it wasn't like that some years ago but if I am correct now it is. If so, then this should be fine I think?

Reply:
We recently had a user that had Node but didn't have NPM: https://github.com/wasp-lang/wasp/issues/1980#issuecomment-2184011331

The problem isn't necessarily that it isn't installed. It might just not be in the `PATH`.

Reply:
Here's the issue: https://github.com/wasp-lang/wasp/issues/2324